kubernetes/samples/scripts/domain-lifecycle/helper.sh

```diff
@@ -249,13 +249,56 @@ function createPatchJsonToUpdateClusterPolicy {
   local policy=$3
   local __result=$4
   
-  replacePolicyCmd="(.spec.clusters[] | select (.clusterName == \"${clusterName}\") \
-    | .serverStartPolicy) |= \"${policy}\""
-  startPolicy=$(echo ${domainJson} | jq "${replacePolicyCmd}" | jq -cr '(.spec.clusters)')
+  clusters=$(echo ${domainJson} | jq -cr '(.spec.clusters)')
+  if [ "${clusters}" == "null" ]; then
+    # cluster doesn't exist, add cluster with server start policy
+    addClusterStartPolicyCmd=".[.| length] |= . + {\"clusterName\":\"${clusterName}\", \
+      \"serverStartPolicy\":\"${policy}\"}"
+    startPolicy=$(echo ${clusters} | jq -c "${addClusterStartPolicyCmd}")
+  else
+    mapCmd="\
+      . |= (map(.clusterName) | index (\"${clusterName}\")) as \$idx | \
+      if \$idx then \
+      .[\$idx][\"serverStartPolicy\"] = \"${policy}\" \
+      else .+  [{clusterName: \"${clusterName}\" , serverStartPolicy: \"${policy}\"}] end"
+    startPolicy=$(echo ${clusters} | jq "${mapCmd}")
+  fi
```

Comment:
Why is this change here? It looks like it relates to 'start policy' not 'replicas', but this pull is all about replicas.  Maybe a bad merge?

Reply:
Also, there's a lot of vars here that should be declared local...

Reply:
It's not a bad merge. If the cluster name is not under `.spec.clusters` for the `stopCluster.sh` or `startCluster.sh` script then it'll not work correctly. Previously I was assuming that cluster will always be present under `.spec.clusters`. Since we removed that assumption for the `scaleCluster.sh` script, I'm making a similar change for `stopCluster.sh` and `startCluster.sh` scripts.

Reply:
OK. Should this be back-ported?

Reply:
SOA team has tested the cluster start/stop scripts and didn't run into any issues with the existing version in 3.1.4. My understanding is that it's an extremely rare use-case to not have any cluster defined under `.spec.clusters` in domain resource. There's a workaround to add the cluster with IF_NEEDED serverStartPolicy. IMO it's not a very high priority to backport since SOA team is also planning to move to 3.2.0 release in May/June.

Reply:
> My understanding is that it's an extremely rare use-case to not have any cluster defined under .spec.clusters in domain resource. 

I think it depends on whether this is during in development or production. I suspect it's the more common case in development.